js/auth.js

```diff
@@ -0,0 +1,306 @@
+    passwordError.classList.remove("show");
+  }
+  checkLoginButton();
+};
+
+const SignupPasswordFocusout = () => {
+  if (signupPassword.value === "") {
+    signupPassword.classList.add("empty_input");
+    signupPasswordError.textContent = "비밀번호를 입력해주세요";
+    signupPasswordError.classList.add("show");
+  } else if (signupPassword.value.length < 8) {
+    signupPassword.classList.add("empty_input");
+    signupPasswordError.textContent = "비밀번호를 8자 이상 입력해주세요";
+    signupPasswordError.classList.add("show");
+  } else {
+    signupPassword.classList.remove("empty_input");
+    signupPasswordError.classList.remove("show");
+  }
+  checkSignupButton();
+};
+
+const NicknameFocusout = () => {
+  if (signupNickname.value === "") {
+    signupNickname.classList.add("empty_input");
+    nicknameError.textContent = "닉네임을 입력해주세요";
+    nicknameError.classList.add("show");
+  } else {
+    signupNickname.classList.remove("empty_input");
+    nicknameError.classList.remove("show");
+  }
+  checkSignupButton();
+};
+
+const PasswordCheckFocusout = () => {
+  if (signupPassword.value !== signupPasswordCheck.value) {
+    signupPasswordCheck.classList.add("empty_input");
+    signupPasswordCheckError.textContent = "비밀번호가 일치하지 않습니다";
+    signupPasswordCheckError.classList.add("show");
+  } else {
+    signupPasswordCheck.classList.remove("empty_input");
+    signupPasswordCheckError.classList.remove("show");
+  }
+  checkSignupButton();
+};
+
+// 버튼 상태 체크 함수
+const checkLoginButton = () => {
+  const hasEmailError = emailError.classList.contains("show");
+  const hasPasswordError = passwordError.classList.contains("show");
+  const isEmailEmpty = loginEmail.value === "";
+  const isPasswordEmpty = loginPassword.value === "";
+
+  if (isEmailEmpty || isPasswordEmpty || hasEmailError || hasPasswordError) {
+    loginButton.disabled = true;
+  } else {
+    loginButton.disabled = false;
+  }
+};
+
+const checkSignupButton = () => {
+  const NicknameError = nicknameError.classList.contains("show");
+  const hasEmailError = signupEmailError.classList.contains("show");
+  const hasPasswordError = signupPasswordError.classList.contains("show");
+  const hasPasswordCheckError =
+    signupPasswordCheckError.classList.contains("show");
+  const isNicknameEmpty = nicknameError.classList.contains("show");
+  const isEmailEmpty = signupEmail.value === "";
+  const isPasswordEmpty = signupPassword.value === "";
+  const isPasswordCheckEmpty = signupPasswordCheck.value === "";
+
+  if (
+    isNicknameEmpty ||
+    isEmailEmpty ||
+    isPasswordEmpty ||
+    isPasswordCheckEmpty ||
+    NicknameError ||
+    hasEmailError ||
+    hasPasswordError ||
+    hasPasswordCheckError
+  ) {
+    signupButton.disabled = true;
+  } else {
+    signupButton.disabled = false;
+  }
+};
+
+const LoginButtonClick = (e) => {
+  e.preventDefault();
+  if (!loginButton.disabled) {
+    console.log("로그인 시도:", loginEmail.value);
+    window.location.href = "../pages/items.html";
+  }
```

Comment:
💊 제안  
button이 disabled 상태의 경우 동작이 되지 않으므로 관련 코드는 없어도 될 것 같아요~

```suggestion
  console.log("로그인 시도:", loginEmail.value);
  window.location.href = "../pages/items.html";
```